components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/api/model/User.java

```diff
@@ -27,24 +28,90 @@
  */
 public class User {
 
-    private String id;
+    private final String id;
     private final List<UserClaim> claims =  new ArrayList<>();
+    private final List<String> groups = new ArrayList<>();
+    private final List<String> roles = new ArrayList<>();
 
     public User(String id) {
 
         this.id = id;
     }
 
+    public User(Builder builder) {
+
+        this.id = builder.id;
+        this.claims.addAll(builder.claims);
+        this.groups.addAll(builder.groups);
+        this.roles.addAll(builder.roles);
+    }
+
     public String getId() {
 
         return id;
     }
 
-    public void setClaims(UserClaim userClaim) {
+    public List<UserClaim> getClaims() {
+
+        return Collections.unmodifiableList(claims);
+    }
+
+    public void addClaim(UserClaim userClaim) {
+
         claims.add(userClaim);
     }
```

Comment:
IMO we should not give this add claim in this POJO. It's rather immutable right ?
So this should come to the builder.
It's only the authentication extension using this right